xrootd/protocol/protocol/protocol.go

```diff
@@ -0,0 +1,153 @@
+	// IsServer indicates whether this server has manager role.
+	IsManager Flags = 0x00000002
+	// IsServer indicates whether this server has meta attribute.
+	IsMeta Flags = 0x00000100
+	// IsServer indicates whether this server has proxy attribute.
+	IsProxy Flags = 0x00000200
+	// IsServer indicates whether this server has supervisor attribute.
+	IsSupervisor Flags = 0x00000400
+)
+
+// SecurityOptions are the security-related options.
+// See specification for details: http://xrootd.org/doc/dev45/XRdv310.pdf, p. 72.
+type SecurityOptions byte
+
+const (
+	// None specifies that no security options are provided.
+	None SecurityOptions = 0
+	// ForceSecurity specifies that signing is required even if the authentication
+	// protocol does not support generic encryption.
+	ForceSecurity SecurityOptions = 0x02
+)
+
+// SecurityInfoLength is the length of SecurityInfo in bytes.
+const SecurityInfoLength = 6
+
+// SecurityInfo is a response part that is provided when required (if server supports that).
+// It contains security version, security options, security level, and number of following security overrides, if any.
+type SecurityInfo struct {
+	// FIXME: Rename Reserved* fields to _ when automatically generated (un)marshalling will be available.
+	Reserved1             byte
+	Reserved2             byte
+	SecurityVersion       byte
+	SecurityOptions       SecurityOptions
+	SecurityLevel         SecurityLevel
+	SecurityOverridesSize byte
+}
+
+// SecurityLevel is the predefined security level that specifies which requests should be signed.
+// See specification for details: http://xrootd.org/doc/dev45/XRdv310.pdf, p. 75.
+type SecurityLevel byte
+
+const (
+	// NoneLevel indicates that no request needs to be signed.
+	NoneLevel SecurityLevel = 0
+	// Compatible indicates that only potentially destructive requests need to be signed.
+	Compatible SecurityLevel = 1
+	// Standard indicates that potentially destructive requests
+	// as well as certain non-destructive requests need to be signed.
+	Standard SecurityLevel = 2
+	// Intense indicates that request that may reveal metadata or modify data need to be signed.
+	Intense SecurityLevel = 3
+	// Pedantic indicates that all requests need to be signed.
+	Pedantic SecurityLevel = 4
+)
+
+// RequestLevel is the security requirement that the associated request is to have.
+type RequestLevel byte
+
+const (
+	// SignNone indicates that the request need not to be signed.
+	SignNone RequestLevel = 0
+	// SignLikely indicates that the request must be signed if it modifies data.
+	SignLikely RequestLevel = 1
+	// SignNeeded indicates that the request mush be signed.
+	SignNeeded RequestLevel = 2
+)
+
+// SecurityOverrideLength is the length of SecurityOverride in bytes.
+const SecurityOverrideLength = 2
+
+// SecurityOverride is a alteration needed to the specified predefined security level.
+// It consist of the request index and the security requirement that associated request is to have.
```

Comment:
`s/It consist/It consists/`
`s/that associated/the associated/`
`s/is to have/should have/`